cel/README.md

```diff
@@ -0,0 +1,221 @@
+
+# Common Expression Language
+
+[![License](https://img.shields.io/badge/License-Apache%202.0-blue.svg)](https://github.com/kubernetes/experimental/blob/master/LICENSE)
+
+This is an experimental project that provides support for Common Expression Language (CEL) in Tekton Pipelines.
+The functionality is provided by a controller that implements the Custom Task interface. Its use cases include 
+evaluating complex expressions to be used in [`WhenExpressions`](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#guard-task-execution-using-whenexpressions)
+in subsequent `Tasks` to guard their execution. 
+
+- [Install](#install)
+- [Usage](#usage)
+- [Uninstall](#uninstall)
+
+## Install
+
+Install and configure `ko`
+
+```
+ko apply -f celcontroller.yaml
+```
+
+This will build and install the controller on your cluster, in the namespace `cel`
+
+## Usage
+
+Two resources are required to evaluate a `CEL` expression using a `Custom Task` are:
+- `CELTask` which defines the API version and identifies the resource object as `CEL`
```

Comment:
Unless we expand on this work some more, we don't actually need to define a `CELTask` CRD type, we can just require that `Run`s that evaluate a CEL expression have the apiVersion+kind, and no `name`, and a `param` named `expression`.

Reply:
yes yes, updated the docs